internal/settings/settings_test.go

```diff
@@ -40,3 +43,40 @@ func TestDecodeOptions_success(t *testing.T) {
 		t.Fatalf("options mismatch: %s", diff)
 	}
 }
+
+func TestValidate_IgnoreDirectoryNames_error(t *testing.T) {
+	tables := []struct {
+		input  string
+		result string
+	}{
+		{datadir.DataDirName, `cannot ignore data directory ".terraform"`},
+		{filepath.Join("path", "path"), fmt.Sprintf(`expected directory name, got a path: %q`, filepath.Join("path", "path"))},
+	}
+
+	for _, table := range tables {
+		out, err := DecodeOptions(map[string]interface{}{
+			"IgnoreDirectoryNames": []string{table.input},
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		result := out.Options.Validate()
+		if result.Error() != table.result {
+			t.Fatalf("expected error: %s, got: %s", table.result, result)
+		}
+	}
+}
+func TestValidate_IgnoreDirectoryNames_success(t *testing.T) {
+	out, err := DecodeOptions(map[string]interface{}{
+		"IgnoreDirectoryNames": []string{"directory"},
```

Comment:
```suggestion
		"ignoreDirectoryNames": []string{"directory"},
```
We may want to use the field tag names, to exercise the mapping in the test too.